cmd/commands/image.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"errors"
+	"github.com/projectriff/riff/pkg/core"
+	"github.com/spf13/cobra"
+)
+
+func Image() *cobra.Command {
+	return &cobra.Command{
+		Use:   "image",
+		Short: "Interact with docker images",
+	}
+}
+
+func ImageRelocate(c *core.Client) *cobra.Command {
+	options := core.RelocateImageOptions{}
+
+	command := &cobra.Command{
+		Use:   "relocate",
+		Short: "Relocate docker image names to another registry",
+		Long: `Relocate either a single YAML file or a riff manifest and its YAML files so that image names refer to
+another (private or public) registry.
+
+To transform a single YAML file, use the '--yaml' flag to specify the path or URL of the YAML file. Use the '--output'
+flag to specify the path to contain the transformed YAML file. If '--output' is an existing directory, the YAML file
+will be created in that directory. Otherwise the YAML file will be written to the path specified in '--output'.
+
+To transform a manifest, use the '--manifest' flag to specify the path of a manifest file which provides the paths or
+URLs of the YAML definitions of riff components. Use the '--output' flag to specify the path of a directory to contain
+the transformed manifest and YAML definitions.
+
+Specify the registry hostname using the '--registry' flag, the user owning the images using the '--registry-user' flag,
+and a complete list of the images to be mapped using the '--images' flag. The '--images' flag contains the path of animage manifest file with contents of the following form:
```

Comment:
I did not expect this command to have to specify the whole list of images to take care of (which would be constructed how?). I assumed it would be smart enough to detect image coordinates (this must have been a misunderstanding during our last call, when I talked about regex, etc)

Reply:
I want this command to be automatic and reliable, so rather than matching with a regex, it relies on an accurate list being created separately. I imagine us creating such an image manifest and including it in release archives, for example. See https://github.com/projectriff/riff/issues/742.

Reply:
After discussion seems that indeed to be 100% reliable, we'd rather have a strict image manifest. This doesn't preclude us from having a "smart" tool that can construct / help construct such a manifest, as a separate process